### DISCUSSION TITLE
How to configure Vercel - Serverless function - maxDuration?

### DISCUSSION THREAD

**Discussion by alatalo** [nuxt/nuxt#24005] [Questions]
I am running a Nuxt 3.7.4 project in Vercel using serverless functions. I'm on a Pro plan and trying to configure the serverless function's max duration timeout to be more than the default 15 seconds Vercel is having.

`nuxt.config.js` is using nitro preset 'vercel':

```ts
// nuxt.config.js
  nitro: {
    preset: 'vercel',
  },
```

Trying to configure the maxDuration according to the [documentation](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) in `vercel.json`:

```json
{
  "functions": {
    "server/api/foo.ts": {
      "maxDuration": 300
    }
  }
}
```

..but no matter what I put in the function path, I get the following error when building:

```sh
$ vercel build
Vercel CLI 32.5.0
Error: The pattern "server/api/foo.ts" defined in `functions` doesn't match any Serverless Functions inside the `api` directory.
Learn More: https://vercel.link/unmatched-function-pattern
```

The file is there, in the defined path:

```sh
$ ls -l server/api/foo.ts
-rw-r--r--  1 bofh  staff  135  28 Oct 07:56 server/api/foo.ts
```

Any ideas how to get the Vercel's maxDuration configuration working on Nuxt 3 or any pointers to what I am doing wrong here? Thanks!

**Comment by oclockTwo** [comment]
Hi, I'm also experiencing this problem, have you solved it yet?

**Comment by alatalo** [comment]
@oclockTwo Nope, could not figure it out. We moved to Vercel edge functions instead and refactored all our backend API endpoints to streaming. No timeout problems anymore. Original issue remains unsolved :/

**Reply by oclockTwo** [reply]
I found the correct configuration yesterday in the discard community. I've tested it. It's working.

export default defineNuxtConfig({
  nitro: {
    vercel: {
      functions: {
        maxDuration: 300,
      }
    }
  },
});

**Reply by alatalo** [reply]
Okay, cool! Could be added to Nuxt and/or Vercel documentation somewhere maybe :)

**Comment by quecot** [comment]
Hi,
I have a similar question: how can I configure functions to includeFiles in serverless runtime?

I've also gone the route of trying to configure vercel.json like explained [here](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#examples-of-reading-files):
![image](https://github.com/nuxt/nuxt/assets/45916187/7af3cba5-d66a-41ba-8215-c1dcc72e8920)

Testing your solution, I can't find a way to configure that property:
![image](https://github.com/nuxt/nuxt/assets/45916187/d115af25-0839-4f1e-8e12-2ea580c0e9cf)

I've created a minimal reproduction of what I'm trying to achieve in this [repo](https://gitlab.com/queco/serverless-include-files/).

Many thanks!

**Reply by quecot** [reply]
Since I posted this issue, the Vercel documentation on this has been updated and it's now resolved:
https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-nuxt
